provider/cmd/app/app.go

```diff
@@ -497,22 +500,30 @@ func runMetricsServer(ctx context.Context, setupLog logr.Logger, opts HTTPServer
 		IdleTimeout:  opts.IdleTimeout,
 	}
 
+	var wg sync.WaitGroup
+	wg.Add(1)
```

Comment:
Since only one goroutine is being spawned, there's no need for the wg WaitGroup. Adding wg.Add(1) and wg.Wait() for a single goroutine introduces unnecessary complexity.

Reply:
how do you mean that? I don't want to close parent goroutine before end goroutine for shutdown.

Reply:
ok. I agree with you.  If the WaitGroup is not used to wait for this goroutine to finish, the runMetricsServer function could return before the metrics server is properly shut down, potentially leaving the server running in the background.